test/Microsoft.AspNet.Razor.Test/Parser/Html/HtmlDocumentTest.cs

```diff
@@ -70,7 +70,11 @@ public void ParseDocumentCorrectlyHandlesOddlySpacedHTMLElements()
             ParseDocumentTest("<div ><p class = 'bar'> Foo </p></div >",
                 new MarkupBlock(
                     BlockFactory.MarkupTagBlock("<div >"),
-                    BlockFactory.MarkupTagBlock("<p class = 'bar'>"),
+                    new MarkupTagBlock(
+                        Factory.Markup("<p"),
+                        new MarkupBlock(
+                            Factory.Markup(" class")),
+                        Factory.Markup(" = 'bar'>")),
```

Comment:
From a non-TagHelper perspective: It'll be treated as equal. Technically there'll be more blocks in the mix but the output will remain unchanged.

From a TagHelper perspective: You'll actually see the code in the HTML elements now. Before it was ignored completely. Aka: `<input checked class="btn" />` resulted in an output.Attributes with 1 attribute (the checked wasn't even acknowledged); poorly formed attributes would either disrupt parsing of other correct attributes or be ignored entirely.